File: taloa-web/types/owner.ts
import type { TagStatus } from "@/types/tag";

export interface LastScan {
  scanned_at: string | null;
  location_lat: number | null;
  location_lng: number | null;
  location_granted: boolean;
}

export interface TagInfo {
  tag_code: string;
  status: TagStatus;
  tag_type: string | null;
}

export interface HealthAlert {
  title: string;
  record_type: string;
  next_due_date: string;
  days_until: number; // negativo = expirado
}

export interface PetSummary {
  id: string;
  name: string;
  species: string;
  photo_url: string | null;
  breed_or_morph: string | null;
  sex: string | null;
  age_years: number | null;
  date_of_birth: string | null;
  colour: string | null;
  microchip: string | null;
  allergies: string | null;
  medication: string | null;
  behaviour: string | null;
  public_notes: string | null;
  emergency_notes: string | null;
  vet_name: string | null;
  vet_phone: string | null;
  show_phone: boolean;
  show_email: boolean;
  // campos por tag_type (Etapa 19)
  travel_notes: string | null;
  airline_approved: boolean | null;
  habitat_temp_min: number | null;
  habitat_temp_max: number | null;
  feeding_schedule: string | null;
  handling_notes: string | null;
  lighting_notes: string | null;
  humidity_notes: string | null;
  critical_conditions: string | null;
  critical_medication: string | null;
  blood_type: string | null;
  tag: TagInfo | null;
  last_scan: LastScan | null;
  health_alert: HealthAlert | null;
}

export interface FoundReportSummary {
  id: string;
  tag_code: string | null;
  pet_name: string | null;
  found_area: string | null;
  notes: string | null;
  finder_phone: string | null;
  created_at: string | null;
}

export interface OwnerInfo {
  name: string | null;
  email: string | null;
}

export interface PawTransaction {
  points: number;
  reason: string;
  created_at: string | null;
}

export interface PawPointsSummary {
  total: number;
  total_earned: number;
  transactions: PawTransaction[];
}

export interface Dashboard {
  owner: OwnerInfo;
  pets: PetSummary[];
  pending_found_reports: FoundReportSummary[];
  paw_points: PawPointsSummary;
}

export interface PetUpdatePayload {
  name?: string;
  species?: string;
  breed_or_morph?: string | null;
  sex?: string | null;
  age_years?: number | null;
  date_of_birth?: string | null;
  colour?: string | null;
  microchip?: string | null;
  photo_url?: string | null;
  allergies?: string | null;
  medication?: string | null;
  behaviour?: string | null;
  public_notes?: string | null;
  emergency_notes?: string | null;
  vet_name?: string | null;
  vet_phone?: string | null;
  show_phone?: boolean;
  show_email?: boolean;
  // campos por tag_type (Etapa 19)
  travel_notes?: string | null;
  airline_approved?: boolean | null;
  habitat_temp_min?: number | null;
  habitat_temp_max?: number | null;
  feeding_schedule?: string | null;
  handling_notes?: string | null;
  lighting_notes?: string | null;
  humidity_notes?: string | null;
  critical_conditions?: string | null;
  critical_medication?: string | null;
  blood_type?: string | null;
}

File: taloa-api/app/schemas/owner.py
"""Schemas do dashboard do dono e edicao de pet."""
from pydantic import BaseModel


class OwnerInfo(BaseModel):
    name: str | None = None
    email: str | None = None


class TagInfo(BaseModel):
    tag_code: str
    status: str
    tag_type: str | None = None


class LastScan(BaseModel):
    scanned_at: str | None = None
    location_lat: float | None = None
    location_lng: float | None = None
    location_granted: bool = False


class PetSummary(BaseModel):
    id: str
    name: str
    species: str
    photo_url: str | None = None
    breed_or_morph: str | None = None
    sex: str | None = None
    age_years: int | None = None
    date_of_birth: str | None = None
    colour: str | None = None
    microchip: str | None = None
    # campos do perfil (para pre-preencher a edicao)
    allergies: str | None = None
    medication: str | None = None
    behaviour: str | None = None
    public_notes: str | None = None
    emergency_notes: str | None = None
    vet_name: str | None = None
    vet_phone: str | None = None
    show_phone: bool = True
    show_email: bool = False
    # campos por tag_type (Etapa 19)
    travel_notes: str | None = None
    airline_approved: bool | None = None
    habitat_temp_min: float | None = None
    habitat_temp_max: float | None = None
    feeding_schedule: str | None = None
    handling_notes: str | None = None
    lighting_notes: str | None = None
    humidity_notes: str | None = None
    critical_conditions: str | None = None
    critical_medication: str | None = None
    blood_type: str | None = None
    tag: TagInfo | None = None
    last_scan: LastScan | None = None


class FoundReportSummary(BaseModel):
    id: str
    tag_code: str | None = None
    pet_name: str | None = None
    found_area: str | None = None
    notes: str | None = None
    finder_phone: str | None = None
    created_at: str | None = None


class PawTransaction(BaseModel):
    points: int
    reason: str
    created_at: str | None = None


class PawPointsSummary(BaseModel):
    total: int = 0
    total_earned: int = 0
    transactions: list[PawTransaction] = []


class DashboardResponse(BaseModel):
    owner: OwnerInfo
    pets: list[PetSummary]
    pending_found_reports: list[FoundReportSummary]
    paw_points: PawPointsSummary = PawPointsSummary()


class PetUpdate(BaseModel):
    name: str | None = None
    species: str | None = None
    breed_or_morph: str | None = None
    sex: str | None = None
    age_years: int | None = None
    date_of_birth: str | None = None
    colour: str | None = None
    microchip: str | None = None
    photo_url: str | None = None
    # perfil
    allergies: str | None = None
    medication: str | None = None
    behaviour: str | None = None
    public_notes: str | None = None
    emergency_notes: str | None = None
    vet_name: str | None = None
    vet_phone: str | None = None
    show_phone: bool | None = None
    show_email: bool | None = None
    # campos por tag_type (Etapa 19)
    travel_notes: str | None = None
    airline_approved: bool | None = None
    habitat_temp_min: float | None = None
    habitat_temp_max: float | None = None
    feeding_schedule: str | None = None
    handling_notes: str | None = None
    lighting_notes: str | None = None
    humidity_notes: str | None = None
    critical_conditions: str | None = None
    critical_medication: str | None = None
    blood_type: str | None = None


class ActionResponse(BaseModel):
    ok: bool = True
    status: str | None = None
